routes/login.js

```diff
@@ -0,0 +1,99 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const config = require("../config");
+const jwtService = require("../services/JwtService");
+
+const userService = require("../services/UserService");
+
+router.get("/", (req, res) => {
+    userService
+        .getAllUsers()
+        .then((users) => {
+            res.json({ users: users });
+        })
+        .catch(() => {
+            res.json({ msg: "no user" });
+        });
+});
+
+router.get("/verify", (req, res) => {
+    const user = jwtService.verify(req.headers["authorization"]);
+    user != undefined
+        ? res.status(200).json({ msg: "verified" })
+        : res.status(401).json({ msg: "unauthorized" });
+});
+
+router.post("/login", (req, res) => {
+    const { zprn, password } = req.body;
+    if (zprn === undefined || password === undefined) {
+        res.status(401).json({ msg: "unauthorized" });
+    } else {
+        userService
+            .getUser(zprn)
+            .then((user) => {
+                if (user != undefined) {
+                    bcrypt.compare(password, user.password, (error, match) => {
+                        if (error) {
+                            res.status(500).json({ msg: "Error occured" });
+                        } else {
+                            if (match) {
+                                const token = jwtService.sign(user.department);
+                                token != null
+                                    ? res.status(200).json({ token: token })
+                                    : res.status(500).json({ msg: "Error!" });
+                            } else {
+                                res.status(401).json({
+                                    msg: "Password doesn't match!",
+                                });
+                            }
+                        }
+                    });
+                } else {
+                    res.status(200).json({ msg: "Not Found!" });
```

Comment:
Does this `"Not Found!"` refer to user not found? If that's the case let's keep all our messages the same way. Also it sounds too ambiguous.

Reply:
Yes, it sounds ambiguous, but I did this for testing purpose, would surely change this!